src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/BidirectionStreamingTest.cs

```diff
@@ -22,7 +22,9 @@ public BidirectionStreamingTest(ITestOutputHelper output) : base(output)
 
         // Build number suggested by the WinHttp team.
         // It can be reduced if bidirectional streaming is backported.
-        public static bool OsSupportsWinHttpBidirectionalStreaming => Environment.OSVersion.Version >= new Version(10, 0, 22357, 0);
+        public static bool OsSupportsWinHttpBidirectionalStreaming => Environment.OSVersion.Version >= new Version(10, 0, 22357, 0)
+            || Environment.OSVersion.Version == new Version(10, 0, 20348, 0); // This is required for WS2022
+                                                                              // because WINHTTP_FLAG_AUTOMATIC_CHUNKING is backported to WS2022.
```

Comment:
I suppose we can't lower the requirement to just >= 10,0,20348 because that would break on other OSes?

Reply:
> I suppose we can't lower the requirement to just >= 10,0,20348 because that would break on other OSes?

I tried, and it broke Win11 21H2 I think in CI.
Log file: [Win11](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-104722-merge-1b216b4d2dba4d16a3/System.Net.Http.WinHttpHandler.Functional.Tests/3/console.6436c32e.log?helixlogtype=result)

Reply:
What I am fearing is that this will break again once the WS 2022 version changes

Reply:
Build version doesn't change, revision changes only but it's not getting reported through `Environment.OSVersion.Version` somehow, but what I can do is, I can check something like this:
```c#
Environment.OSVersion.Version >= new Version(10, 0, 20348, 0) && Environment.OSVersion.Version < new Version(10, 0, 20349, 0);
```
So if something changes in the future and WS2022 reports any revision number, it will not break.

Reply:
looking at current PlatformDetection.Windows code, WS 2022 "version" has not changed in 2 years, so it is probably fine to keep the code as is.

But I would consider adding PlatformDetection.IsWindowsServer2022 to hide this complex check behind something readable.

Reply:
> But I would consider adding PlatformDetection.IsWindowsServer2022 to hide this complex check behind something readable.

Makes sense, will do

Reply:
My concern here is that this test will start to fail if the WS2022 version is below than 20348.2527 in local runs, because `Environment.OSVersion.Version` doesn't return `Revision` property properly.
What do you think about this?